### PR TITLE
Add documentation for date_optional_time formatter (elastic#50457)

### LINKED ISSUE #50457: Incomplete documentation on strict_date_optional_time format
**Describe the feature**: Explain more how the `strict_date_optional_time` format.

Today, the documentation here: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html#strict-date-time
is only  "A generic ISO datetime parser where the date is mandatory and the time is optional."

There is no format description like the other ones. I had to check in the code what it is exactly: 
https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/common/time/DateFormatters.java#L107

### PR DESCRIPTION
**Pull request by arkel-s**
Precise the documentation about date_optional_time format.
Closes elastic#50457

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @arkel-s, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in your Git [commit](https://github.com/elastic/elasticsearch/pull/50458.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

File: docs/reference/mapping/params/format.asciidoc
Comment (jrodewig) [inline docs/reference/mapping/params/format.asciidoc]:
```suggestion
    mandatory and the time is optional.
```

Based on my tests, including a `T` in the input will result in an error. However, the following format worked:

```
[source,console]
--------------------------------------------------
PUT my_index
{
  "mappings": {
    "properties": {
      "date": {
        "type":   "date",
        "format": "strict_date_optional_time || yyyy-MM-dd hh:mm:ss.SSSZ"
      }
    }
  }
}
--------------------------------------------------
```

Also: `separated` is misspelled.

File: docs/reference/mapping/params/format.asciidoc
Comment (jrodewig) [inline docs/reference/mapping/params/format.asciidoc]:
```suggestion
    Examples: `yyyy-MM-dd hh:mm:ss.SSSZ` or  `yyyy-MM-dd`.
```

**Review by jrodewig — CHANGES_REQUESTED** [review]
Thanks for submitting this PR @arkel-s.

I left some feedback to address. Once that's take care of, I can help merge these changes in.

**Comment by jrodewig** [comment]
@elasticmachine test this please

File: docs/reference/mapping/params/format.asciidoc
Comment (rjernst) [reply docs/reference/mapping/params/format.asciidoc]: @jrodewig The `T` is definitely required, when a time is specified. What was the date you tested with?

Comment (jrodewig) [reply docs/reference/mapping/params/format.asciidoc]: Hi @rjernst,

You're right. The `T` is required for the index API. My initial attempts were using the create index API, but after reviewing the docs, I think I just misinterpreted what was there.

My apologies @arkel-s . I'll clean up my review suggestions here.

**Comment by jrodewig** [comment]
@arkel-s This should be good to go once the `separated` misspelling is addressed.

**Comment by arkel-s** [comment]
Thanks @jrodewig, typo fixed!

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks again @arkel-s!

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
### Backport commits
master: f20367e40581bad40f281b8d59382dc94a9aa13c
7.x: d5f4790
7.5: 1fbd028

**Pull request merged**
